homeassistant/components/eyeonwater/coordinator.py

```diff
@@ -0,0 +1,40 @@
+"""EyeOnWater coordinator."""
+import logging
+
+from pyonwater import Account, Client, EyeOnWaterAPIError, EyeOnWaterAuthError, Meter
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EyeOnWaterData:
```

Comment:
Creating your own DataUpdateCoodrinator will avoid adding fetch logic in your `__init__.py`